source/lac/petsc_matrix_free.cc

```diff
@@ -72,9 +70,8 @@ namespace PETScWrappers
                          const unsigned int n,
                          const unsigned int local_rows,
                          const unsigned int local_columns)
-    : communicator(MPI_COMM_WORLD)
   {
-    do_reinit(m, n, local_rows, local_columns);
+    do_reinit(MPI_COMM_WORLD, m, n, local_rows, local_columns);
```

Comment:
I think it is better to use `MPI_COMM_WORLD` to be consistent with the rest of the code that uses `MPI_` communicators.

Reply:
So, which way do we go? `MPI_COMM_WORLD` or `PETSC_COMM_WORLD`?

Reply:
Sorry for the slow response - in principle we should always use `PETSC_COMM_WORLD` here since PETSc may not be set up to use `MPI_COMM_WORLD`, but since we use `MPI_COMM_WORLD` all over the place we should make that change in one place later.